pytests/test_client.py

```diff
@@ -0,0 +1,272 @@
+# License: MIT
+# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+
+"""Tests for the methods in the client."""
+import asyncio
+import unittest
+from datetime import datetime, timedelta
+from decimal import Decimal
+from unittest.mock import AsyncMock, MagicMock
+
+from frequenz.api.electricity_trading.v1 import electricity_trading_pb2
+from frequenz.client.electricity_trading import (
+    Client,
+    Currency,
+    DeliveryArea,
+    DeliveryPeriod,
+    Energy,
+    EnergyMarketCodeType,
+    MarketActor,
+    MarketSide,
+    Order,
+    OrderDetail,
+    OrderExecutionOption,
+    OrderState,
+    OrderType,
+    Price,
+    StateDetail,
+    StateReason,
+    TradeState,
+)
+from google.protobuf import timestamp_pb2
+
+
+class TestClient(unittest.TestCase):  # pylint: disable=too-many-instance-attributes
+    """Test suite for the Client class."""
+
+    def setUp(self) -> None:
```

Comment:
(just for the record): Because this is a function inheriting from `unittest.TestCase`, but this won't matter anyways because I'm changing to `pytest` now